### PR TITLE
Support for UCX 1.10+

### PR DESCRIPTION
**Pull request by pentschev**
Select correct transports for UCX 1.10+ and ensure the CUDA context is created before initializing UCX when support for that is requested.

### CODE REVIEW

**Comment by pentschev** [comment]
cc @quasiben

**Comment by pentschev** [comment]
I don't know the cause of the failing tests but I don't think they are related to this PR, could someone confirm/deny?

File: distributed/comm/ucx.py [hunk 1]
```diff
@@ -53,13 +53,22 @@ def init_once():
...
+    ucx_config = _scrub_ucx_config()
+    if "TLS" in ucx_config and "cuda_copy" in ucx_config["TLS"]:
+        try:
+            import numba.cuda
+        except ImportError:
+            raise ImportError(
+                "CUDA support with UCX requires Numba for context management"
+            )
+
+        numba.cuda.current_context()
```
Comment (quasiben) [inline distributed/comm/ucx.py:66]: I think we should put a comment here explaining why we need to create a cuda context before starting communications

Comment (pentschev) [reply distributed/comm/ucx.py:70]: Done in https://github.com/dask/distributed/pull/4787/commits/7c199af218e3808bf6e7d144986d4e951aeb134b

Comment (quasiben) [reply distributed/comm/ucx.py:70]: Thanks @pentschev !

**Comment by pentschev** [comment]
Failure is:

```python
_______________________________ test_thread_time _______________________________

    def test_thread_time():
        start = metrics.thread_time()
        run_for(0.05)
        dt = metrics.thread_time() - start
>       assert 0.03 <= dt <= 0.2
E       assert 0.03 <= 0.020078000000012253

distributed/tests/test_metrics.py:48: AssertionError
```

Seems like a very tight timeout. Not sure what's the general policy, should we increase it slightly to `0.3` or ignore it for the time being?

**Comment by quasiben** [comment]
I think we should raise that as an issue but it's safe to ignore for this PR

**Comment by pentschev** [comment]
Makes sense, I raised https://github.com/dask/distributed/issues/4790 for that.

**Comment by quasiben** [comment]
This also failed on `distributed/tests/test_nanny.py::test_num_fds FAILED ` and but only on mac 3.7 but also due to a timeout.  I added to issue #4790.  I think it's safe to merge this PR

**Pull request merged**